doc/content/xapi/internals/generated.md

```diff
@@ -0,0 +1,345 @@
++++
+title = "Generated Parts of Xapi"
++++
+
+## Introduction
+
+Many parts of `xapi` are auto-generated during the build process.
+This article aims to document some of these modules and how they relate to each
+other. The intention of this article is to serve as a developer resource and,
+as such, its contents are prone to change (or become inaccurate) as the
+codebase evolves. The ultimate source of truth remains the codebase itself.
+
+## Interface Description
+
+All of XenAPI's data model is described within the `ocaml/idl` subdirectory.
```

Comment:
Not to be confused with `xapi-idl`, which describes the internal APIs in a different form